src/main/java/org/apache/commons/compress/compressors/lha/BinaryTree.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.commons.compress.compressors.lha;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import org.apache.commons.compress.compressors.CompressorException;
+import org.apache.commons.compress.utils.BitInputStream;
+import org.apache.commons.lang3.ArrayFill;
+
+/**
+ * Binary tree of positive values.
+ *
+ * Copied from org.apache.commons.compress.archivers.zip.BinaryTree and modified for LHA.
+ */
+class BinaryTree {
+
+    /** Value in the array indicating an undefined node */
+    private static final int UNDEFINED = -1;
+
+    /** Value in the array indicating a non leaf node */
+    private static final int NODE = -2;
+
+    /**
+     * The array representing the binary tree. The root is at index 0, the left children (0) are at 2*i+1 and the right children (1) at 2*i+2.
+     */
+    private final int[] tree;
+
+    /**
+     * Constructs a binary tree from the given array that contains the depth (code length) in the
+     * binary tree as values in the array and the index into the array as the value of the leaf node.
+     *
+     * If the array contains a single value, this is a special case where there is only one node in the
+     * tree (the root node) and it contains the value. For this case, the array contains the value of
+     * the root node instead of the depth in the tree. This special case also means that no bits will
+     * be read from the bit stream when the read method is called, as there are no children to traverse.
+     *
+     * @param array the array to build the binary tree from
+     */
+    BinaryTree(final int[] array) {
```

Comment:
Use a vararg declaration here and call sites won't have to explicitly declare an array each time.